src/component/handlers/edit/editOnPaste.js

```diff
@@ -31,15 +32,23 @@ import type {EntityMap} from 'EntityMap';
 /**
  * Paste content.
  */
-function editOnPaste(editor: DraftEditor, e: SyntheticClipboardEvent): void {
-  e.preventDefault();
-  var data = new DataTransfer(e.clipboardData);
+function editOnPaste(editor: DraftEditor, e: DOMEvent): void {
+  // e in this case is a native DOM event, instead of a SyntheticClipboardEvent,
+  // because react doesn't support capture of paste, and the bubbling event triggers
+  // when the event is at #document
+  // For the pasteTrap to work, either we need to be capturing, or e.currentTarget needs to be
+  // the contentEditable div. So, e in this case comes from a direct editor.addEventListener
+  // Therefore, we need to replicate anything that the SyntheticClipboardEvent does that is used
+  // Currently, that is only getting the clipboard, which involves falling back to window for IE & Edge.
+  const clipboard = e.clipboardData ? e.clipboardData : window.clipboardData;
```

Comment:
Minor: I haven't looked but does the code use this pattern or `const clipboard = e.clipboardData || window.clipboardData`?